### PR TITLE
spec: tweak spec file based on Fedora review

### PR DESCRIPTION
**Pull request by siteshwar**
Related: https://bugzilla.redhat.com/show_bug.cgi?id=2275304

### CODE REVIEW

**Comment by siteshwar** [comment]
Shall I bump up the version number on this change?

File: osh.spec [hunk 1]
```diff
@@ -6,6 +6,7 @@ Release:        1%{?dist}
 License:        GPL-3.0-or-later
 Summary:        Static and Dynamic Analysis as a Service
 URL:            https://github.com/openscanhub/openscanhub/
+# Sources are available at https://github.com/openscanhub/openscanhub/tags
```
Comment (siteshwar) [inline osh.spec:9]: The full URL is removed by PackIt, so I am leaving a comment here.

**Comment by kdudka** [comment]
> Shall I bump up the version number on this change?

No.

**Review by kdudka — CHANGES_REQUESTED** [review]
Per https://bugzilla.redhat.com/show_bug.cgi?id=2275304#c2 you should also remove `Conflicts: covscan-client < %{version}`.

The comments about the description were IMO not related to the top-level `osh` package, which is non-existent for end users, but about all the subpackages.

File: osh.spec [hunk 2]
```diff
@@ -31,6 +32,7 @@ BuildRequires:  systemd-rpm-macros
 %description
 OpenScanHub is a service for static and dynamic analysis.
 It consists of central hub, workers and cli client.
+It runs various static (and dynamic) analyzers on RPM packages.
```
Comment (kdudka) [inline osh.spec:35]: I find the description a bit clumsy.  The first and last sentence duplicate each other.  Cannot you just append ` of RPM packages` to the first sentence?

File: osh.spec [hunk 3]
```diff
@@ -75,7 +77,8 @@ Recommends: osh-worker-conf
 Obsoletes: covscan-worker < %{version}
 
 %description worker
-OpenScanHub worker
+OpenScanHub worker.
+It is used to perform scans.
```
Comment (kdudka) [inline osh.spec:81]: I am not sure if this is an improvement really.  I would say that all the subpackages are used "to perform scans".  I would reuse the description we already have in `kobo`: https://src.fedoraproject.org/rpms/kobo/blob/3b18f70fbe4809a4aa7be89b6c93b3d716738b60/f/kobo.spec#_73

File: osh.spec [hunk 4]
```diff
@@ -47,10 +49,10 @@ Provides: osh-cli = %{version}-%{release}
 # This is kept here for backward compatibility with old package name
 Provides: covscan-client = %{version}
 Obsoletes: covscan-client < %{version}
-Conflicts: covscan-client < %{version}
 
 %description client
-OpenScanHub CLI client
+OpenScanHub CLI client.
+It is used to submit, query and download builds.
```
Comment (kdudka) [inline osh.spec:55]: `builds` -> `scans`

**Review by kdudka — CHANGES_REQUESTED** [review]

**Review by kdudka — APPROVED** [review]

File: osh.spec [hunk 5]
```diff
@@ -232,7 +231,6 @@ fi
 %attr(640,root,root) %config(noreplace) %{_sysconfdir}/osh/worker.conf
 
 %files hub
-%defattr(-,root,apache,-)
```
Comment (xsuchy) [inline osh.spec:235]: If you remove this then files will be owned by root:root.
I do NOT know what should be ownership of the files. It just seemed suspicious to me that all files in this subpackage are owned by root:apache.

Comment (siteshwar) [reply osh.spec:235]: Tests in [Testing Farm](https://artifacts.dev.testing-farm.io/1b0bc17a-a514-46d3-8aab-7c9d051b6350/) passed, so I believe these changes should not break existing deployments.

Comment (kdudka) [reply osh.spec:235]: @xsuchy I was double checking this before the approval.  The files that need to be in the `apache` group use `%attr` explicitly.  Everything else is just Python scripts and configuration files, where the group can be set to `root`.

**Pull request merged**